src/csharp/Grpc.Core.Tests/ChannelCredentialsTest.cs

```diff
@@ -52,6 +58,9 @@ public void ChannelCredentials_NativeCredentialsAreReused()
             var nativeCreds1 = creds.ToNativeCredentials();
             var nativeCreds2 = creds.ToNativeCredentials();
             Assert.AreSame(nativeCreds1, nativeCreds2);
+
+            var nativeCreds3 = ChannelCredentials.Secure.ToNativeCredentials();
```

Comment:
the native creds are only going to be  identical if the original SslCredentials instance is the same instance (which in your case is not)

I think you need this:
```
var nativeCreds3 = ChannelCredentials.Secure.ToNativeCredentials();
var nativeCreds4 = ChannelCredentials.Secure.ToNativeCredentials();
Assert.AreSame(nativeCreds3, nativeCreds4);
```

Reply:
Fixed